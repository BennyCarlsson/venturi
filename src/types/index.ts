export enum FontWeight {
  normal = 400,
  medium = 500,
  bold = 700
}

export type Token = {
  accessToken: string;
  expires: null | Date;
};

export type ApiError = {
  DepartureBoard: {
    error: string;
    errorText: string;
    noNamespaceSchemaLocation: string;
  };
};

export type DepartureBoard = {
  DepartureBoard: {
    Departure: Departure[];
    noNamespaceSchemaLocation: string;
    serverdate: string;
    servertime: string;
  };
};

export type Departure = {
  JourneyDetailRef: { ref: string };
  accessibility: string;
  bgColor: string;
  date: string;
  direction: string;
  fgColor: string;
  journeyNumber: string;
  journeyid: string;
  name: string;
  rtDate: string;
  rtTime: string;
  sname: string;
  stop: string;
  stopid: string;
  stroke: string;
  time: string;
  track: string;
  type: string;
};
