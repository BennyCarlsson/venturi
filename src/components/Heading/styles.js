import styled, { css } from 'styled-components'
import Colors from 'tokens/colors'
import { FontWeight as _fontWeight } from 'types/index'

const generalFontStyle = ({ fontSize, color, fontWeight, center, uppercase }) =>
  css`
    font-size: ${fontSize ? `${fontSize}px` : 'initial'};
    color: ${color ? color : Colors.vBlack};
    font-weight: ${fontWeight ? fontWeight : _fontWeight.normal};
    text-align: ${center ? 'center' : 'initial'};
    text-transform: ${uppercase ? 'uppercase' : 'none'};
  `

export const StyledH1 = styled.h1`
  ${(props) => generalFontStyle(props)};
`
export const StyledH2 = styled.h2`
  ${(props) => generalFontStyle(props)};
`
export const StyledH3 = styled.h3`
  ${(props) => generalFontStyle(props)};
`
export const StyledH4 = styled.h4`
  ${(props) => generalFontStyle(props)};
`
export const StyledH5 = styled.h5`
  ${(props) => generalFontStyle(props)};
`
export const StyledH6 = styled.h6`
  ${(props) => generalFontStyle(props)};
`
