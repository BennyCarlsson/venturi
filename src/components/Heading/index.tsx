import React from 'react';
import { FontWeight } from 'types';
import { StyledH1, StyledH2, StyledH3, StyledH4, StyledH5, StyledH6 } from './styles';

export enum HeadingType {
  H1 = 'h1',
  H2 = 'h2',
  H3 = 'h3',
  H4 = 'h4',
  H5 = 'h5',
  H6 = 'h6'
}

type HeadingProps = {
  children: React.ReactNode;
  fontSize?: number;
  fontWeight?: FontWeight;
  color?: string;
  center?: boolean;
  uppercase?: boolean;
  headingType: HeadingType;
};

const Heading = ({ headingType, children, ...props }: HeadingProps) => {
  switch (headingType) {
    case HeadingType.H1:
      return <StyledH1 {...props}>{children}</StyledH1>;
    case HeadingType.H2:
      return <StyledH2 {...props}>{children}</StyledH2>;
    case HeadingType.H3:
      return <StyledH3 {...props}> {children}</StyledH3>;
    case HeadingType.H4:
      return <StyledH4 {...props}>{children}</StyledH4>;
    case HeadingType.H5:
      return <StyledH5 {...props}>{children}</StyledH5>;
    case HeadingType.H6:
      return <StyledH6 {...props}>{children}</StyledH6>;
    default:
      return <StyledH1 {...props}>{children}</StyledH1>;
  }
};

export default Heading;
