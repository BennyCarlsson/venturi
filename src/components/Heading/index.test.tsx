import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Heading, { HeadingType } from '.';
import { FontWeight } from 'types';

describe('<Heading />', () => {
  it('should match snapshot', () => {
    const tree = renderer
      .create(
        <Heading
          headingType={HeadingType.H1}
          fontSize={16}
          fontWeight={FontWeight.bold}
          color="green"
          center
        >
          test
        </Heading>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should renders without crashing', () => {
    render(
      <Heading
        headingType={HeadingType.H1}
        fontSize={16}
        fontWeight={FontWeight.bold}
        color="green"
        center
      >
        testy test
      </Heading>
    );
    expect(screen).toBeTruthy();
    expect(screen.queryByText('testy test')).toBeInTheDocument();
  });

  const headingTypes = Object.values(HeadingType);
  headingTypes.forEach((headingType) => {
    it(`${headingType} should render passed child element`, () => {
      const text = 'Testy test';
      render(<Heading headingType={headingType}>{text}</Heading>);
      expect(screen.getByText(text, { selector: headingType })).toBeInTheDocument();
    });
  });
});
