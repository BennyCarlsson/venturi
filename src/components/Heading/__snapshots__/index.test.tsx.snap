// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should match snapshot 1`] = `
<h1
  className="sc-bdfBwQ gdNazK"
  color="green"
  fontSize={16}
  fontWeight={700}
>
  test
</h1>
`;
